### PR TITLE
Refactor markdown files

### PR DESCRIPTION
**Pull request by Bituvo**
**Ready for review**

This is a better version of #13503, but without so much noise. Only markdown files (and worlds_here.txt) were changed.

This PR adds these things to the documentation:
 - Modern primary and secondary headings
 - Capitalized headers
 - Slightly less whitespace before primary headers (three lines)
 - Github flavored warnings and information boxes
 - Lua code blocks where appropriate
 - Tables where appropriate
 - Inline code blocks where appropriate
 - Removal of extra whitespace in tables (`{ min=3, max=5 }` -> `{min=3, max=5}`)
 - Typo fixes (especially `id` instead of `ID`)
 - Removed table of contents from `/README.md` as Github has a builtin one
 - And more

To do (preferably in a different PR, just putting it out there):
 - Use a consistent maximum line width (maybe increasing with each level of bullet points)
 - Scour for typos and other mistakes
 - Redo `protocol.txt`

Some people still use the plaintext version of the Lua API, so I tried to maintain readability while still simplifying things.

### CODE REVIEW

**Comment by appgurueu** [comment]
First, an observation: This PR does not require a core dev to review it. I suggest the @minetest/docs team (or anyone trustworthy, really) be allowed to review this.

> Github flavored warnings and information boxes

I recently (regrettably) pushed such a change; I'll have to fix it (or you could fix it here too, if you want). This seems to not render correctly on https://api.minetest.net. I think we want to stick to plain Markdown, for now.

**Comment by Bituvo** [comment]
> > Github flavored warnings and information boxes
> 
> I recently (regrettably) pushed such a change; I'll have to fix it (or you could fix it here too, if you want). This seems to not render correctly on https://api.minetest.net. I think we want to stick to plain Markdown, for now.

That website uses MkDocs, which supports extensions. There are already a few extensions enabled [here](https://github.com/minetest/minetest/blob/master/doc/mkdocs/build.sh#L14) and after some searching I found https://pypi.org/project/markdown-gfm-admonition/, which enables Github admonition boxes. I personally haven't tried testing this extension (I am too busy), but it looks promising, to whoever may be able to try.

**Comment by Bituvo** [comment]
@Zughy Whose opinion should I get on this?

**Comment by Zughy** [comment]
It was about the GH info boxes but I guess it's been solved

**Comment by Bituvo** [comment]
Who should I ask about the extension?

**Comment by Bituvo** [comment]
I've added a prerequisite PR with the necessary changes.

**Comment by p-ouellette** [comment]
I haven't contributed to Minetest in a while so you don't need to give too much weight to my comment, but I disagree with some of these changes.

- I think sentence case should be used for headings, not title case. It's more readable and less complicated to write. See e.g. https://www.tempertemper.net/blog/sentence-case-versus-title-case.
- I prefer dashes for list items over asterisks. IMO dashes look cleaner, they're easier to type, and `*` is not vertically centered in most fonts.

**Comment by Bituvo** [comment]
> * I think sentence case should be used for headings, not title case. It's more readable and less complicated to write. See e.g. https://www.tempertemper.net/blog/sentence-case-versus-title-case.

That blog post doesn't really apply to this kind of markdown documentation, especially because the words used in headings are usually just a simple word or phrase. The blog also makes a point that sentence case is useful for when a word could change the meaning of a sentence depending on its capitalization, which doesn't apply to any heading here.

> * I prefer dashes for list items over asterisks. IMO dashes look cleaner, they're easier to type, and `*` is not vertically centered in most fonts.

I do believe that this is a matter of opinion (in which mine is to use asterisks), but after some searching, I found [Google's markdown style guide](https://google.github.io/styleguide/docguide/style.html), which describes what I've been doing to the documentation almost to a point. Hyphens are never used in unordered lists here.

I do appreciate the concern! Please tell me if there is anything else that could/should be changed in your opinion. I want the documentation to be readable and accessible by everyone who reads and uses it, not just me.

File: doc/developing/misc.md
Comment (Bituvo) [reply doc/developing/misc.md:12]: It was to stay consistent with the other codeblocks. But, now that you mention it, the Minetest code style guidelines say to use tabs... I'll do it tomorrow.

Comment (Bituvo) [reply doc/developing/misc.md:12]: This is turning out to be harder than I expected.

File: doc/lua_api.md
Comment (Bituvo) [reply doc/lua_api.md:1]: Done!

**Comment by Bituvo** [comment]
About the new ruling proposed in https://gist.github.com/celeron55/bf93a47442e418a629181908f68ffb0f:
- This PR only touches markdown files, not code or networking stuff
- This PR may or may not be simple, depending on who you ask, but I argue that it is and it certainly won't break anything
- Of course this PR passes, because it doesn't change any code :)

File: doc/lua_api.md
Comment (sfan5) [inline doc/lua_api.md:2377]: this is pseudo-code so we can keep the concise syntax
```suggestion
    damage += cap.damage_groups[group]
```

File: doc/lua_api.md
Comment (sfan5) [inline doc/lua_api.md:2700]:
```suggestion
> Enabled automatically when using formspec version 2 or higher.
```

File: doc/lua_api.md
Comment (sfan5) [inline doc/lua_api.md:4143]: seems to be missing `>` here

File: doc/lua_api.md
Comment (sfan5) [inline doc/lua_api.md:6503]:
```suggestion
> `on_rightclick` is not called when the wielded item overrides `on_place`
```

File: doc/lua_api.md
Comment (sfan5) [inline doc/lua_api.md:6645]: accidental content removal

File: doc/lua_api.md
Comment (sfan5) [inline doc/lua_api.md:6777]: forgot to delete the copy

File: doc/lua_api.md
Comment (sfan5) [inline doc/lua_api.md:6997]:
```suggestion
* `minetest.parse_json(string[, nullvalue])`: returns parsed data
```

File: doc/lua_api.md
Comment (sfan5) [inline doc/lua_api.md:7012]:
```suggestion
> JSON is stricter than the Lua table format:
```

File: doc/lua_api.md
Comment (sfan5) [inline doc/lua_api.md:7537]: this isn't any clearer, just different.

File: doc/lua_api.md
Comment (sfan5) [inline doc/lua_api.md:7541]: 32 bits is "single precision"

File: doc/lua_api.md
Comment (sfan5) [inline doc/lua_api.md:7796]: rebase mistake

File: doc/lua_api.md
Comment (sfan5) [inline doc/lua_api.md:7707]: I thought function names were supposed to be in code formatting?

File: doc/lua_api.md
Comment (sfan5) [inline doc/lua_api.md:7810]: missing `{`
also it's more readable with spacing

File: doc/lua_api.md
Comment (sfan5) [inline doc/lua_api.md:7908]: content accidentally removed.

File: doc/lua_api.md
Comment (sfan5) [inline doc/lua_api.md:8158]: "the ColorSpec"
which?

File: doc/lua_api.md
Comment (sfan5) [inline doc/lua_api.md:8274]: if the default is 1.0 then `[0,1)` must be incorrect

File: doc/lua_api.md
Comment (sfan5) [inline doc/lua_api.md:8411]: rebase mistake

File: doc/lua_api.md
Comment (sfan5) [inline doc/lua_api.md:8859]: context lost

**Review by sfan5 — CHANGES_REQUESTED** [review]

File: doc/android.md
Comment (Bituvo) [reply doc/android.md]: Removed the Requirements section.

File: doc/android.md
Comment (Bituvo) [reply doc/android.md]: Removed keystore point.

File: doc/client_lua_api.md
Comment (Bituvo) [reply doc/client_lua_api.md:201]: Must have been a remnant from other merges, removed

File: doc/compiling/linux.md
Comment (Bituvo) [reply doc/compiling/linux.md]: Removed this line.

File: doc/compiling/macos.md
Comment (Bituvo) [reply doc/compiling/macos.md:5]: How did that happen... fixed

File: doc/compiling/windows.md
Comment (Bituvo) [reply doc/compiling/windows.md:48]: Converted to lowercase inline codeblocks.

File: doc/world_format.md
Comment (Bituvo) [reply doc/world_format.md:473]: Removed note.

File: util/README_mod_translation_updater.md
Comment (Bituvo) [reply util/README_mod_translation_updater.md:41]: Added suggestion.

File: doc/lua_api.md
Comment (Bituvo) [reply doc/lua_api.md:87]: Re-added the word.

File: doc/lua_api.md
Comment (Bituvo) [reply doc/lua_api.md]: Fixed section.

File: doc/lua_api.md
Comment (Bituvo) [reply doc/lua_api.md:251]: Oops, moved back

File: doc/lua_api.md
Comment (Bituvo) [reply doc/lua_api.md:777]: Added suggestion.

File: doc/lua_api.md
Comment (Bituvo) [reply doc/lua_api.md:2340]: Removed.

File: doc/lua_api.md
Comment (Bituvo) [reply doc/lua_api.md:2376]: Added suggestion.

File: doc/lua_api.md
Comment (Bituvo) [reply doc/lua_api.md:2698]: Added suggestion.

File: doc/lua_api.md
Comment (Bituvo) [reply doc/lua_api.md:3726]: Added suggestion.

File: doc/lua_api.md
Comment (Bituvo) [reply doc/lua_api.md:4141]: True, added them

File: doc/lua_api.md
Comment (Bituvo) [reply doc/lua_api.md:6501]: Added suggestion.

File: doc/lua_api.md
Comment (Bituvo) [reply doc/lua_api.md:6645]: Re-added content

File: doc/lua_api.md
Comment (Bituvo) [reply doc/lua_api.md]: Fixed notes here

File: doc/lua_api.md
Comment (Bituvo) [reply doc/lua_api.md]: Added suggestion.

File: doc/lua_api.md
Comment (Bituvo) [reply doc/lua_api.md:7000]: Added suggestion.

File: doc/lua_api.md
Comment (Bituvo) [reply doc/lua_api.md:7529]: Removed both bit rewordings.

File: doc/lua_api.md
Comment (Bituvo) [reply doc/lua_api.md:7796]: Re-added section.

File: doc/lua_api.md
Comment (Bituvo) [reply doc/lua_api.md:7700]: You are right, changed

File: doc/lua_api.md
Comment (Bituvo) [reply doc/lua_api.md:7803]: Added back the `{`, but I will not be putting the spacing back. This is one of the points of this PR

File: doc/lua_api.md
Comment (Bituvo) [reply doc/lua_api.md:7901]: Re-added.

File: doc/lua_api.md
Comment (Bituvo) [reply doc/lua_api.md:8151]: Specified `sky_color`.

File: doc/lua_api.md
Comment (Bituvo) [reply doc/lua_api.md:8267]: Fixed this section.

File: doc/lua_api.md
Comment (Bituvo) [reply doc/lua_api.md:8411]: Re-added.

File: doc/lua_api.md
Comment (Bituvo) [reply doc/lua_api.md]: Fixed context.

**Comment by Bituvo** [comment]
@sfan5 Thank you so much for digging through this pull. I would never have the time or energy to do it myself, so help is always appreciated. I hope you agree with my changes.

**Comment by sfan5** [comment]
Should be fine, can you merge/rebase the last tiny change?

File: doc/lua_api.md
Comment (sfan5) [inline doc/lua_api.md:2958]: also missing `>` here and above

File: doc/lua_api.md
Comment (sfan5) [inline doc/lua_api.md:2975]: and here

File: doc/lua_api.md
Comment (sfan5) [inline doc/lua_api.md:4431]: and here

File: doc/lua_api.md
Comment (sfan5) [inline doc/lua_api.md:7677]: and here

File: doc/lua_api.md
Comment (Bituvo) [reply doc/lua_api.md:2958]: I searched for `\n> .*\n[^>*] ` and resolved all of these issues that I could find.

**Comment by Bituvo** [comment]
Merge conflicts have been resolved.

**Comment by Bituvo** [comment]
@sfan5 With all due respect, this pull request has been open since before I finished middle school. It has received a few reviews from you and corpserot, but no talk has been made on whether or not this is realistically going to be merged, especially due to the simplicity of this PR. I've done my best to try and organize the changes here to make it easy to review. Is it time to get a final opinion on the matter?

**Comment by sfan5** [comment]
> It has received a few reviews from you and corpserot, but no talk has been made on whether or not this is realistically going to be merged, especially due to the simplicity of this PR

I do not have time atm to dig through changes of this size.
Despite the simplicity it needs a careful review to make sure mistakes are not accidentally introduced.

If nobody else picks this up consider it closed. Sorry.
(You could join IRC and ask if any other dev wants to pick it up)

**Comment by Zughy** [comment]
as discussed on IRC, reviewing a PR this big is no easy task - and nobody seems interested in doing it. Closing

**Comment by Bituvo** [comment]
Ok, I'll try to separate it into multiple PRs.

**Pull request closed without merging**
